### PR TITLE
feat: add repr methods to interaction data classes

### PR DESCRIPTION
**Pull request by onerandomusername**
## Summary

add repr methods to interaction data classes

## Checklist

- [ ] If code changes were made, then they have been tested
    - [ ] I have updated the documentation to reflect the changes
    - [x] I have formatted the code properly by running `task lint`
    - [x] I have type-checked the code by running `task pyright`
- [ ] This PR fixes an issue
- [x] This PR adds something new (e.g. new method or parameters)
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: disnake/interactions/application_command.py [hunk 1]
```diff
@@ -306,6 +312,12 @@ def __init__(
         ]
         self.focused: bool = data.get("focused", False)
 
+    def __repr__(self):
+        return (
+            f"<ApplicationCommandInteractionDataOption name={self.name!r} type{self.type!r} "
```
Comment (shiftinv) [inline disnake/interactions/application_command.py:317]:
```suggestion
            f"<ApplicationCommandInteractionDataOption name={self.name!r} type={self.type!r} "
```

File: disnake/interactions/modal.py [hunk 2]
```diff
@@ -134,3 +134,6 @@ def __init__(self, *, data: ModalInteractionDataPayload):
         self.custom_id: str = data["custom_id"]
         # this attribute is not meant to be used since it lacks most of the component data
         self._components: List[ActionRow] = [ActionRow(d) for d in data["components"]]
+
+    def __repr__(self):
+        return "<ModalInteractionData custom_id={self.custom_id} _components={self._components}>"
```
Comment (shiftinv) [inline disnake/interactions/modal.py:139]:
```suggestion
        return f"<ModalInteractionData custom_id={self.custom_id!r} _components={self._components!r}>"
```

File: disnake/interactions/message.py [hunk 3]
```diff
@@ -133,3 +133,6 @@ def __init__(self, *, data: ComponentInteractionDataPayload):
         self.custom_id: str = data["custom_id"]
         self.component_type: ComponentType = try_enum(ComponentType, data["component_type"])
         self.values: Optional[List[str]] = data.get("values")
+
+    def __repr__(self):
+        return f"<MessageInteractionData custom_id={self.custom_id!r} component_type={self.component_type!r}>"
```
Comment (shiftinv) [inline disnake/interactions/message.py:138]: Should this contain `self.values` as well?

**Review by shiftinv — CHANGES_REQUESTED** [review]

File: disnake/interactions/application_command.py (hunk 1 above)
Comment (onerandomusername) [reply disnake/interactions/application_command.py:317]: also missing `>`

**Review by shiftinv — APPROVED** [review]

**Pull request merged**